server/src/main/java/io/druid/server/QueryResource.java

```diff
@@ -85,14 +86,17 @@ public QueryResource(
       QuerySegmentWalker texasRanger,
       ServiceEmitter emitter,
       RequestLogger requestLogger,
-      QueryManager queryManager
+      QueryManager queryManager,
+      InjectableValues guiceInjectableValues
   )
   {
     this.config = config;
     this.jsonMapper = jsonMapper.copy();
+    this.jsonMapper.setInjectableValues(guiceInjectableValues); // Workaround for https://github.com/FasterXML/jackson-databind/issues/696
```

Comment:
After investigating the code and running a few tests, there doesn't seem to be any side effect of using the injected ObjectMappers. I'm uploading a new commit which removes the copy. I'll change the PR title when that is up.